src/map/ai/ai_mob_dummy.cpp

```diff
@@ -438,8 +438,8 @@ void CAIMobDummy::ActionDropItems()
                     }
                 }
 
-                //check for gil (beastmen drop gil, some NMs drop gil)
-                if(m_PMob->CanDropGil() || map_config.all_mobs_gil_bonus > 0)
+                // check for gil (beastmen drop gil, some NMs drop gil)
+                if (m_PMob->CanDropGil() || map_config.all_mobs_gil_bonus > 0 && m_PMob->getMobMod(MOBMOD_GIL_MAX) >= 0)) // Negative value of MOBMOD_GIL_MAX is used to prevent gil drops in Dynamis/Limbus.

```

Comment:
when combining && and || in a statement, you should put braces to make the intention clear